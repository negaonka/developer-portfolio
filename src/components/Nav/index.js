import React, {useEffect} from 'react';
import { capitalizeFirstLetter } from '../../utils/helper';

function Nav(props){

    const {
        menus = [],
        setCurrentMenu,
        currentMenu
      } = props;

    useEffect(() => {
        document.title = capitalizeFirstLetter(currentMenu.name);
      }, [currentMenu]);

 return (

      <nav>
        <ul className="flex-row">
        {menus.map((menu) => (
            <li
              className={`mx-2 ${
                currentMenu.name === menu.name && 'navActive'
                }`}
              key={menu.name}
            >
              <span
                onClick={() => {
                  setCurrentMenu(menu);
                }}
              >
                {capitalizeFirstLetter(menu.name)}
              </span>
            </li> 
            ))}
        </ul>
      </nav>
 )
}

export default Nav;