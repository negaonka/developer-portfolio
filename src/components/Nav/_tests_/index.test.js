import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

const menus = [
    { name: 'about me'}
  ]
  const mockCurrentMenu = jest.fn();
  const mockSetCurrentMenu = jest.fn();

afterEach(cleanup);  //to clean all the memory data

describe('Nav component', () => {
    // baseline test
    it('renders', () => {
        render(<Nav 
            menus={menus}
            setCurrentMenu={mockSetCurrentMenu}
            currentMenu={mockCurrentMenu}/>);
      });
   
    // snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Nav   
            menus={menus}
            setCurrentMenu={mockSetCurrentMenu}
            currentMenu={mockCurrentMenu}/>);
        // assert value comparison
        expect(asFragment()).toMatchSnapshot();
      });

    })