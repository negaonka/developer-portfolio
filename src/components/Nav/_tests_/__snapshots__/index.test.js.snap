// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="flex-row"
    >
      <li
        class="mx-2 false"
      >
        <span>
          About me
        </span>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
